### PR TITLE
Use Generic ListView

### PR DESCRIPTION
**Pull request by sks444**
Closes https://github.com/coala/community/issues/87

### CODE REVIEW

File: community/urls.py [hunk 1]
```diff
@@ -92,4 +99,39 @@ def get_index():
...
+    ),
+    distill_url(
+        r'^viewmodel/contributors/$',
+        ContributorListView.as_view(),
+        name='contributors',
+        distill_func=get_index,
+        distill_file='viewmodel/templates/contributor_list.html',
+    ),
+    distill_url(
+        r'^viewmodel/contributor/(?P<pk>.*)/$',
```
Comment (sks444) [inline community/urls.py]:
```
django.urls.exceptions.NoReverseMatch: Reverse for 'contributordetail' with no arguments not found. 1 pattern(s) tried: ['viewmodel/contributor/(?P<pk>.*)/$']
```
@jayvdb, why netlify is giving this error, it is working locally.

File: openhub/models.py [hunk 2]
```diff
@@ -92,14 +105,20 @@ class OutsideCommitter(models.Model):
...
+
 
 class MostCommit(models.Model):
 
     project = models.CharField(max_length=100)
     commits = models.IntegerField()
 
     def __str__(self):
-        return self.project1
+        return self.project
```
Comment (jayvdb) [inline openhub/models.py]: separate issue for this.  it is a bug, which needs to be fixed immediately, not as part of this change.  it should be labelled low, and advertise it on gsoc room for someone to fix it for you.

File: templates/viewmodel/templates/base.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {% endif %}
+  {% endblock %} 
+  
+  
+  </div>
+</div>
+
+</div>
+</body>
+</html>
```
Comment (jayvdb) [inline templates/viewmodel/templates/base.html]: EOL at EOF

File: viewmodel/apps.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class ViewmodelConfig(AppConfig):
+    name = 'viewmodel'
```
Comment (jayvdb) [inline viewmodel/apps.py]: can we call it 'model' and the directory 'model' and the URL path '/models/'

File: templates/viewmodel/templates/portfolio_project_list.html [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{% extends "viewmodel/templates/base.html" %}
+
+{% block content %}
+        <h1>Portfolio Project List</h1>
```
Comment (jayvdb) [inline templates/viewmodel/templates/portfolio_project_list.html]: whooping change indent here.  Add an issue to add a bear to standardise our .html spacing.  Possibly do this one yourself, so it is done quickly.

File: community/urls.py [hunk 6]
```diff
@@ -24,8 +44,61 @@ def get_index():
     return None
 
 
-urlpatterns = [
+def get_all_contributors():
+    # This function needs to return an iterable of dictionaries. Dictionaries
```
Comment (jayvdb) [inline community/urls.py]: We do not need this comment repeated everywhere.  The code is ridiculously simple and obvious.

File: model/views.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    PortfolioProject,
+    OutsideProject,
+    OutsideCommitter,
+    AffiliatedCommitter,
+    Organization
+    )
+
+
+def index(request):
+    args = {'portfolioprojects': 'portfolio_projects',
```
Comment (jayvdb) [inline model/views.py]: always put dict members starting on the next line, and ALWAYS end EVERY row with a trailing comma, and the trailing `}` on a separate line.

File: templates/model.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        <a href="{% url 'affiliatedcommitters' %}">Affiliated Committers</a><br>
+      </li>
+      <li>
+        <a href="{% url 'outsidecommitters' %}">Outside Committers</a><br>
+      </li>
+      <li>
+        <a href="{% url 'organization' %}">Organizations</a><br>
+      </li>
+      <li>
+        <a href="">OpenHub Organizations</a>(Not imported yet)<br>
```
Comment (jayvdb) [inline templates/model.html]: remove

File: templates/model.html [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
+    <title>Visit your models</title>
+  </head>
+  <body>
+    <h1>Visit Your Models and its Imported data</h1>
+    <hr>
+    <div class="container">
+      <h2>data app models</h2>
+      <li>
+        <a href="">Contributors</a>(Not implemented yet)
```
Comment (jayvdb) [inline templates/model.html]: dont include stuff which hasnt been implemented yet

File: templates/model.html [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      <li>
+        <a href="{% url 'organization' %}">Organizations</a><br>
+      </li>
+      <li>
+        <a href="">OpenHub Organizations</a>(Not imported yet)<br>
+      </li>
+    </div>
+    <hr>
+    <div class="container">
+      <h2>gsoc app models</h2>
```
Comment (jayvdb) [inline templates/model.html]: remove

File: templates/model.html [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    </div>
+    <hr>
+    <div class="container">
+      <h2>gci app models</h2>
```
Comment (jayvdb) [inline templates/model.html]: remove

File: templates/model/templates/affiliated_committer_detail.html [hunk 12]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>Visit models</title>
```
Comment (jayvdb) [inline templates/model/templates/affiliated_committer_detail.html]: wrong titles (everywhere)

maybe there is a variable you can use instead?

File: templates/model/templates/affiliated_committer_detail.html [hunk 13]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (jayvdb) [inline templates/model/templates/affiliated_committer_detail.html]: the base layout of these html pages should be one template that other the other templates re-use

File: model/views.py [hunk 14]
```diff
@@ -0,0 +1,76 @@
+from django.shortcuts import render
+from django.views import generic
+
+from openhub.models import (
+    PortfolioProject,
+    OutsideProject,
+    OutsideCommitter,
+    AffiliatedCommitter,
+    Organization
```
Comment (jayvdb) [inline model/views.py]: trailing ,

File: community/urls.py [hunk 15]
```diff
@@ -16,6 +16,26 @@
 from data.views import index as contributors_index
 from inactive_issues.inactive_issues_scraper import inactive_issues_json
 from openhub.views import index as openhub_index
+from model.views import index as model_index
+from openhub.models import (
+    PortfolioProject,
+    OutsideCommitter,
+    AffiliatedCommitter,
+    OutsideProject,
+    Organization
```
Comment (jayvdb) [inline community/urls.py]: trailing ,

File: templates/model/templates/affiliated_committer_list.html [hunk 16]
```diff
@@ -0,0 +1,18 @@
+{% extends "model/templates/base.html" %}
+{% block title %}
+<title>Affiliated Contributors List</title>
+{% endblock %}
+{% block content %}
+<h1>Affiliated Contributors List</h1>
+{% if affiliated_committer_list %}
+<ul>
+  {% for affiliatedcommitter in affiliated_committer_list %}
```
Comment (jayvdb) [inline templates/model/templates/affiliated_committer_list.html]: affiliatedcommitter -> affiliated_committer

(same applies everwhere you are not using snake_case. jinja is based on python ; same naming conventions apply.

File: templates/model/templates/affiliated_committer_detail.html [hunk 17]
```diff
@@ -0,0 +1,15 @@
+{% extends "model/templates/base.html" %}
+{% block title %}
+<title>{{ affiliatedcommitter.name }}</title>
```
Comment (jayvdb) [inline templates/model/templates/affiliated_committer_detail.html]: can this be `affiliated_committer` ?

**Comment by jayvdb** [comment]
ack https://github.com/coala/community/pull/115/commits/b4505bb209420d703e8efbc2ab250fa4c4f95059

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**